app/components/primer/state_component.rb

```diff
@@ -4,12 +4,19 @@ module Primer
   # Component for rendering the status of an item.
   class StateComponent < Primer::Component
     COLOR_DEFAULT = :default
-    COLOR_MAPPINGS = {
+    NEW_COLOR_MAPPINGS = {
+      open: "State--green",
+      closed: "State--red",
+      merged: "State--purple"
+    }.freeze
+
+    DEPRECATED_COLOR_MAPPINGS = {
       COLOR_DEFAULT => "",
       :green => "State--green",
       :red => "State--red",
       :purple => "State--purple"
```

Comment:
I _believe_ this should work:

```suggestion
      open: "State--open",
      closed: "State--closed",
      merged: "State--merged"
    }.freeze

    DEPRECATED_COLOR_MAPPINGS = {
      COLOR_DEFAULT => "",
      green: "State--open",
      red: "State--closed",
      purple: "State--merged"
```

Doing so enables us to fully switch to the functional class names while migrating the consumers of this component to use the functional color names.

Reply:
When I did this locally, the doc examples rendered without color, which told me this was broken. Am I misunderstanding?

Reply:
The problem is that those classes aren't officially released by PrimerCSS and will only work in dotcom.
I think mapping the functional keys to color classes (like `open: "State--green"`) is fine by now. Once Primer releases the new classes we can migrate the classes (without impacting users!)

Reply:
I think we're saying we're go leave this change as is. @joelhawksley is this making sense to you?

Reply:
Ah, I see what's going on here.

The docs examples load Primer from a public CDN, not using the version we serve in dotcom. 

We _really_ need to update the PVC docs site to use the version of Primer CSS we're using on dotcom. Perhaps we could prioritize that ahead of this PR?

Reply:
cool, will follow up in https://github.com/primer/view_components/issues/276

Reply:
@srt32 now I think this change is valid :)